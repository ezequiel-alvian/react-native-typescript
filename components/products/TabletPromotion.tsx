import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import { Dimensions } from "react-native"

const TabletPromotion = ({ promotion } : {promotion:number}) => {
    return(
        <View style={styles.tabletPromotion}>
            <Text style={styles.tabletPromotionText}>{`-${promotion}%`}</Text>
        </View>
    )

}

const styles = StyleSheet.create({
    tabletPromotion: {
        display:'flex',
        justifyContent:'flex-end',
        flexDirection:'column',
        alignItems: 'flex-end',
        width: (Dimensions.get('window').width -40 ),

    },
    tabletPromotionText: {
        padding:5,
        backgroundColor:'#00FF7F',
    }
})

export default TabletPromotion