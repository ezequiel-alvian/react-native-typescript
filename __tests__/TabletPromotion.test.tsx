import TabletPromotion from "../components/products/TabletPromotion"
import { render } from '@testing-library/react-native';


describe('<TabletPromotion/>',()=>{
    let component
    let data = 25
    test('test',()=>{
        component = render(<TabletPromotion promotion={data}/>)
        expect(component.toJSON()).toMatchSnapshot()
      })
})