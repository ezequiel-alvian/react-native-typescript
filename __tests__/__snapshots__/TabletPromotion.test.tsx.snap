// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabletPromotion/> test 1`] = `
<View
  style={
    {
      "alignItems": "flex-end",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "flex-end",
      "width": 710,
    }
  }
>
  <Text
    style={
      {
        "backgroundColor": "#00FF7F",
        "padding": 5,
      }
    }
  >
    -25%
  </Text>
</View>
`;
